### PR TITLE
Test.refactor.coverage

### PR DESCRIPTION
**Pull request by filipecosta90**
to be discussed after #296

### CODE REVIEW

File: opt/system-setup.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def common_first(self):
 
         if self.os == 'linux':
             self.install("ca-certificates")
-        self.install("git cmake unzip wget patchelf awscli")
+        self.install("git cmake unzip wget patchelf awscli valgrind")
```
Comment (rafie) [inline opt/system-setup.py]: This will fail on macOS. Better to install Valgrind in:
```
def linux(self):
    self.install("valgrind")
```

**Comment by rafie** [comment]
This can be way off the current implementation, but as we don't use CMake to run tests, I prefer to drive coverage data collection and report generation outside CMake (the build part is of course ok but it's relatively modest in configuration). TimeSeries has it implemented this way.

**Comment by filipecosta90** [comment]
> This can be way off the current implementation, but as we don't use CMake to run tests, I prefer to drive coverage data collection and report generation outside CMake (the build part is of course ok but it's relatively modest in configuration). TimeSeries has it implemented this way.

Agree @rafie. will check TS and change accordingly :)

**Comment by filipecosta90** [comment]
closing in favor of #299

**Pull request closed without merging**